frappe/email/doctype/newsletter/newsletter.py

```diff
@@ -265,9 +266,10 @@ def get_newsletter_list(doctype, txt, filters, limit_start, limit_page_length=20
 
 def send_scheduled_email():
 	"""Send scheduled newsletter to the recipients."""
-	scheduled_newsletter = frappe.get_all('Newsletter', filters = {
-		'schedule_send': ('<=', now_datetime()),
-		'email_sent': 0
-	}, fields = ['name'])
+	scheduled_newsletter = frappe.db.sql("""
+SELECT name from `tabNewsletter`
+WHERE schedule_send <= %s
+AND email_sent = 0
+""", now_datetime(), as_dict=1)
```

Comment:
```suggestion
	scheduled_newsletter = frappe.get_all('Newsletter', filters = {
		'schedule_send': ('<=', now_datetime()),
		'email_sent': 0
	}, fields = ['name'], ignore_ifnull=True)
```

Test it once.

Reply:
orm adds coalesce where it sets `NULL` to a default value: `0001-01-01`, which is obviously always true for `schedule_send <= now_datetime()`. so anything with no `schedule_send` set and `email_sent=0` will be sent. you don't want that.

Reply:
reference: https://github.com/frappe/frappe/blob/develop/frappe/model/db_query.py#L479-L481

Reply:
also this query should work fine since there is no user input.

Reply:
@Thunderbottom I guess you did not notice `ignore_ifnull`.

You can achieve same thing using ORM.